File: 3.x.x/crates-utopia/span/src/level.rs
//! Severity level of a span (PHP `Utopia\Span\Level`).

/// Names follow Grafana Loki's `detected_level` vocabulary (`warn`, not `warning`).
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
pub enum Level {
    Debug,
    Info,
    Warn,
    Error,
    Fatal,
}

impl Level {
    /// PHP enum string value.
    pub fn as_str(self) -> &'static str {
        match self {
            Self::Debug => "debug",
            Self::Info => "info",
            Self::Warn => "warn",
            Self::Error => "error",
            Self::Fatal => "fatal",
        }
    }

    /// PHP `Level::tryFrom((string) $span->get('level'))`.
    pub fn try_from_attr(value: &str) -> Option<Self> {
        match value {
            "debug" => Some(Self::Debug),
            "info" => Some(Self::Info),
            "warn" => Some(Self::Warn),
            "error" => Some(Self::Error),
            "fatal" => Some(Self::Fatal),
            _ => None,
        }
    }
}

File: packages/span/src/Level.php
<?php

declare(strict_types=1);

namespace Utopia\Span;

/**
 * Severity level of a span, ordered from least to most severe.
 *
 * Names follow Grafana Loki's `detected_level` vocabulary (note `warn`, not
 * `warning`). Exporters translate as needed — e.g. {@see Exporter\Sentry\Level}
 * maps `Warn` to Sentry's `warning`.
 */
enum Level: string
{
    case Debug = 'debug';
    case Info = 'info';
    case Warn = 'warn';
    case Error = 'error';
    case Fatal = 'fatal';
}
